gempy/assets/geographic.py

```diff
@@ -0,0 +1,562 @@
+            self.dip_direction = 360 + np.arctan(self.normal[0] / self.normal[1]) / np.pi * 180.
+
+    #    elif normal_vec[1] == 0:
+    #        return 90
+
+    def stereonet(self):
+        """Create stereonet plot of plane pole and half circle for this point set"""
+        import matplotlib.pyplot as plt
+        import mplstereonet
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='stereonet')
+
+        # ax.plane(dip_dirs, dips, 'g-', linewidth=0.5)
+        ax.pole(self.dip_direction - 90, self.dip, 'gs', markersize=4)
+        ax.plane(self.dip_direction - 90, self.dip, 'g', markersize=4)
+
+        # ax.rake(strike, dip, -25)
+        ax.grid()
+
+    def minmax(self):
+        """Get minimum and maximum values of points in point set (e.g. to determine surrounding box)"""
+        point_array = np.empty((len(self.points), 2))
+
+        for i, p in enumerate(self.points):
+            point_array[i] = (p.x, p.y)
+
+        self.min = np.min(point_array, axis=0)
+        self.max = np.max(point_array, axis=0)
+
+
+class GeoTiffgetValue(object):
+    """Determin a value from a GeoTiff
+
+    Required, for example, to determine z-values in GeoTiffs of DEMs.
+
+    Note: at this stage, the value is taken from Raster Band 1.
+
+    Credits to entry on stackoverflow:
+    http://stackoverflow.com/questions/13439357/extract-point-from-raster-in-gdal
+
+    Args:
+        tifname (filename): path and filename to GeoTiff file
+    """
+
+    def __init__(self, tifname='test.tif'):
+        # open the GeoTiff raster file and its spatial reference
+        self.ds = gdal.Open(tifname)
+        sr_raster = osr.SpatialReference(self.ds.GetProjection())
+
+        # get the WGS84 spatial reference
+        sr_point = osr.SpatialReference()
+        sr_point.ImportFromEPSG(4326)  # WGS84
+
+        # coordinate transformation
+        self.ct = osr.CoordinateTransformation(sr_point, sr_raster)
+
+        # geotranformation and its inverse
+        gt = self.ds.GetGeoTransform()
+        dev = (gt[1] * gt[5] - gt[2] * gt[4])
+        gtinv = (gt[0], gt[5] / dev, -gt[2] / dev,
+                 gt[3], -gt[4] / dev, gt[1] / dev)
+        self.gt = gt
+        self.gtinv = gtinv
+
+        # band as array
+        b = self.ds.GetRasterBand(1)
+        self.arr = b.ReadAsArray()
+
+    def lookup(self, lon, lat):
+        """look up value at lon, lat"""
+
+        # get coordinate of the raster
+        xgeo, ygeo, zgeo = self.ct.TransformPoint(lon, lat, 0)
+
+        # convert it to pixel/line on band
+        u = xgeo - self.gtinv[0]
+        v = ygeo - self.gtinv[3]
+        # FIXME this int() is probably bad idea, there should be
+        # half cell size thing needed
+        xpix = int(self.gtinv[1] * u + self.gtinv[2] * v)
+        ylin = int(self.gtinv[4] * u + self.gtinv[5] * v)
+
+        # look the value up
+        return self.arr[ylin, xpix]
+
+
+class KmlPoints(object):
```

Comment:
I would suggest fiona or geopandas here, as OGR has a kml driver, and anything else you need to do can just be built on top of that

Reply:
Thanks, I'll check that out, that sounds promising!